registration_api.js

```diff
@@ -0,0 +1,596 @@
+                                            target: "#errorDelete",
+                                        },
+                                    },
+                                },
+                                successDelete: {
+                                    id: "successDelete",
+                                    on: {
+                                        REFETCH: "#loadingDelete",
+                                    },
+                                },
+                                errorDelete: {
+                                    id: "errorDelete",
+                                    on: {
+                                        RETRY: "#loadingDelete",
+                                    },
+                                },
+                            },
+                        },
+                        updateApp: {
+                            id: "updateApp",
+                            initial: "loadingUpdate",
+                            states: {
+                                loadingUpdate: {
+                                    id: "loadingUpdate",
+                                    invoke: {
+                                        src: async (_, event) => {
+                                            await appUpdate(event.data)
+                                        },
+                                        onDone: {
+                                            target: "#successUpdate",
+                                        },
+                                        onError: {
+                                            target: "#errorUpdate",
+                                        },
+                                    },
+                                },
+                                successUpdate: {
+                                    id: "successUpdate",
+                                    on: {
+                                        REFETCH: "#loadingUpdate",
+                                    },
+                                },
+                                errorUpdate: {
+                                    id: "errorUpdate",
+                                    on: {
+                                        RETRY: "#loadingUpdate",
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    },
+});
+
+const isStorageSupported = storage => {
+    if (typeof storage === 'undefined') {
+        return false;
+    }
+    const test_key = 'test';
+    try {
+        storage.setItem(test_key, '1');
+        storage.removeItem(test_key);
+        return true;
+    } catch (e) {
+        return false;
+    }
+};
+
+const redirectToLogin = (is_logged_in, language, has_params = true, redirect_delay = 0) => {
+    if (!is_logged_in && isStorageSupported(sessionStorage)) {
+        const l = window.location;
```

Comment:
I think is better to use some different name for variable here

Reply:
This code was copied from a solution in deriv-app. However, valid point, will adjust it